src/sknnr/transformers/_gbnode_transformer.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import annotations
+
+from typing import Literal
+
+import numpy as np
+from numpy.typing import NDArray
+from sklearn.base import BaseEstimator
+from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor
+from sklearn.utils.validation import check_is_fitted
+
+from ._tree_node_transformer import TreeNodeTransformer, uniform_weights
+
+
+def delta_loss(
+    est: GradientBoostingClassifier | GradientBoostingRegressor, X: NDArray, y: NDArray
+) -> NDArray:
+    """
+    Calculate tree weights as a function of the change in loss between
+    successive trees in a gradient boosting estimator.
+    """
+    # See https://github.com/lemma-osu/sknnr/issues/96#issuecomment-2967847215
+    #
+    # Calculate the initial loss, which by default in regression is based
+    # on predicting the mean for all samples. The loss function is
+    # half-squared error, so multiply by 2 to be consistent with how
+    # sklearn calculates loss.
+    initial_loss = (
+        est._loss(np.asarray(y, dtype="float64"), est._raw_predict_init(X)) * 2
```

Comment:
I lazily multiplied the initial loss by 2 to match the default `HalfSquaredError` calculation, but I suspect we should duplicate [the logic used in `sklearn`](https://github.com/scikit-learn/scikit-learn/blob/90338a43e8cd5e22a173ffa56e1fd93e64a08a1d/sklearn/ensemble/_gb.py#L864-L873), just to make sure we don't accidentally end up accidentally doubling the initial loss relative to the other scores.